x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/planning/AssignmentPlan.java

```diff
@@ -315,6 +314,14 @@ private Quality computeQuality() {
         return new Quality(isSatisfyingPreviousAssignments, weighedAllocationsScore, memoryScore);
     }
 
+    public AssignmentPlan withDeploymentsWithZeroAllocations(Collection<Deployment> deploymentsWithZeroAllocations) {
+        Map<Deployment, Map<Node, Integer>> newAssignments = new HashMap<>(assignments);
+        for (Deployment deployment : deploymentsWithZeroAllocations) {
+            newAssignments.put(deployment, Collections.emptyMap());
```

Comment:
nit: This shouldn't happen in general, but maybe you should assert that `deployment` is not in `assignments` already to be defensive. Otherwise, it will be overwritten by an empty map.